main.py

```diff
@@ -877,26 +877,156 @@ def H_ii_indices(det_i: Determinant) -> Iterator[Two_electron_integral_index_pha
             yield (i, j, i, j), 1
 
     @staticmethod
-    def category_C(idx, psi_i, psi_j, N_orb):
+    def get_dets_occ_in_orbitals(
+        spindet_occ: Dict[OrbitalIdx, Set[int]],
+        oppspindet_occ: Dict[OrbitalIdx, Set[int]],
+        d_orbitals: Dict[str, Set[OrbitalIdx]],
+        which_orbitals,
+    ):
         """
-        psi_i psi_j: Psi_det, lists of determinants
-        idx: i,j,k,l, index of integral <ij|kl>
-        For an integral i,j,k,l of category C, yield all dependent determinant pairs (I,J)
-        Possibilities are i = k < j < l: (1,2,1,3), i < k < j = l: (1,3,2,3), j < i = k < l: (2,1,2,3)
+        Get indices of determinants that are occupied in the orbitals d_orbitals.
+        Input which_orbitals = "all" or "any" indicates if we want dets occupied in all of the indices, or just any of the indices
+        """
+        orbitals_occupied = set()
+        for spintype, indices in d_orbitals.items():
+            if spintype == "same":
+                for o in indices:
+                    if len(orbitals_occupied) == 0:
+                        orbitals_occupied = spindet_occ[o]
+                    else:
+                        if which_orbitals == "all":
+                            orbitals_occupied = orbitals_occupied & spindet_occ[o]
+                        else:
+                            orbitals_occupied = orbitals_occupied | spindet_occ[o]
+            if spintype == "opposite":
+                for o in indices:
+                    if len(orbitals_occupied) == 0:
+                        orbitals_occupied = oppspindet_occ[o]
+                    else:
+                        if which_orbitals == "all":
+                            orbitals_occupied = orbitals_occupied & oppspindet_occ[o]
+                        else:
+                            orbitals_occupied = orbitals_occupied | oppspindet_occ[o]
```

Comment:
I think this code can be refactored. Both branches are similar the only difference about `spindet_occ` versus `oppspindet_occ`